text/trace/0173-messaging-semantic-conventions.md

```diff
@@ -0,0 +1,242 @@
+and thus to ship and use stable instrumentation.
+
+## Roadmap
+
+1. This OTEP, consisting of scenarios and a proposed roadmap, is approved and
+   merged.
+2. [Stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability)
+   for semantic conventions are approved and merged. This is not strictly related
+   to semantic conventions for messaging but is a prerequisite for stabilizing any
+   semantic conventions.
+3. OTEPs proposing guidance for general instrumentation problems that also
+   pertain to messaging are approved and merged. Those general instrumentation
+   problems include retries and instrumentation layers.
+4. An OTEP proposing a set of attributes and conventions covering the scenarios
+   in this document is approved and merged.
+5. Proposed specification changes are verified by prototypes for the scenarios
+   and examples below.
+6. The [specification for messaging semantic conventions for tracing](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md)
+   are updated according to the OTEP mentioned above and are declared
+   [stable](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#stable).
+
+## Terminology
+
+To leverage existing standards, the terminology used in this document is based
+on the [CloudEvents specification](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md).
+CloudEvents is hosted by the CNCF and provides a specification for describing
+event data in common formats to provide interoperability across services,
+platforms and systems.
+
+### Message
+
+A "message" is a transport envelope for the transfer of information. The
+information is a combination of a payload and metadata.  Messages may be
+transferred directly between parties or via one or more intermediaries.  A
+message may carry annotations that are directed at intermediaries on the
+message path.  Messages are uniquely identifiable.
+
+In the strict sense, a _message_ is a payload that is sent to a specific
+destination, whereas an _event_ is a signal emitted by a component upon
+reaching a given state. This document is agnostic of those differences and uses
+the term "message" in a wider sense to cover both concepts.
+
+### Producer
+
+The "producer" is a specific instance, process or device that creates and
+publishes a message. "Publishing" is the process of sending a message or batch
+to the intermediary or consumer.
+
+### Consumer
+
+A "consumer" receives the message and acts upon it. It uses the context and
+data to execute some logic, which might lead to the occurrence of new events.
+
+The consumer receives, processes, and settles a message. "Receiving" is the
+process of obtaining a message from the intermediary, "processing" is the
```

Comment:
```suggestion
process of obtaining a message from the intermediary or producer, "processing" is the
```
Since direct communication is possible as well as per the definition in the Message definition above.

Reply:
This was discussed in the workgroup. We decided that the messaging system semantic conventions require at least one intermediary, and I updated the definition of "message" accordingly:

> Messages are transferred via one or more intermediaries.

Other message passing scenarios should be covered by different conventions (refer to the out-of-scope section "Asynchronous message passing in the wider sense").